### PR TITLE
Change/more generated filters

### PR DESCRIPTION
**Pull request by DavidOberholzer**
*Background*: Needed a DateRange filter for some of the new user filter items, and a lot more filters were added to the swagger specification for the user list endpoint.

*What was done*: A dateRangeInput was implemented with the ability to use datetime if required. DateInputs are limited by on another in order to avoid overlapping dates. 

EDIT: Also added in requirement that filters only fire off on 3 characters or more, as the API's return require a minimum of 3 for each filter.

### CODE REVIEW

File: admin/src/inputs/DateRangeInput.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        const { source } = this.props;
+        const today = new Date();
+        const maxDate =
+            this.state.to !== ''
+                ? new Date(this.state.to)
+                : new Date(today.getFullYear() + 100, today.getMonth(), today.getDay());
+        const minDate =
+            this.state.from !== ''
+                ? new Date(this.state.from)
+                : new Date(today.getFullYear() - 100, today.getMonth(), today.getDay());
```
Comment (jwfwessels) [inline admin/src/inputs/DateRangeInput.js]: Would `new Date()` not suffice? aka use the `today` variable above?

File: admin/src/inputs/DateRangeInput.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        return (
+            <span>
+                <Field
+                    name={`${source}.from`}
+                    component={this.component}
+                    props={{
+                        options: {
+                            maxDate: maxDate
```
Comment (jwfwessels) [inline admin/src/inputs/DateRangeInput.js]: 🐒 nit: might be a bit cleaner if you move this obj creation to the top of the render function where you calculate the values... but now I'm just being pedantic 🤔

File: admin/src/inputs/DateRangeInput.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                        }
+                    }}
+                    label="To"
+                    onChange={this.handleToOnChange}
+                />
+            </span>
+        );
+    }
+}
+DateRangeInput.propTypes = {
```
Comment (jwfwessels) [inline admin/src/inputs/DateRangeInput.js]: Dont see where you are using this prop?

File: admin/src/resources/Role.js [hunk 4]
```diff
@@ -29,9 +29,6 @@ const validationCreateRole = values => {
     if (!values.label) {
         errors.label = ["label is required"];
     }
```
Comment (jwfwessels) [inline admin/src/resources/Role.js]: no more 2fa 😞

File: admin/src/swaggerRestServer.js [hunk 5]
```diff
@@ -66,7 +66,14 @@ export const convertRESTRequestToHTTP = ({
 
             if (params.filter) {
                 Object.keys(params.filter).forEach(key => {
-                    query[key] = params.filter[key];
+                    let filter =
+                        params.filter[key] instanceof Object
```
Comment (jwfwessels) [inline admin/src/swaggerRestServer.js]: quite curious as to the need for this check? might be work commenting

File: swagger/management_layer.yml [hunk 6]
```diff
@@ -374,11 +374,11 @@ definitions:
         $ref: '#/definitions/role_label'
```
Comment (jwfwessels) [inline swagger/management_layer.yml]: I assume this file is correct. parsing YAML is only a small step up from parsing XML that is it only gives me a headache as appose to a migraine 🙈

File: admin/src/inputs/DateRangeInput.js (hunk 3 above)
Comment (DavidOberholzer) [reply admin/src/inputs/DateRangeInput.js]: in the constructor

File: swagger/management_layer.yml (hunk 6 above)
Comment (DavidOberholzer) [reply swagger/management_layer.yml]: yup, this is just copy pasting the changes to the management_layer yaml spec and will not generate correctly if not up to swagger specification.

File: admin/src/inputs/DateRangeInput.js (hunk 1 above)
Comment (DavidOberholzer) [reply admin/src/inputs/DateRangeInput.js]: Nope its also a filter input which means they can select further back.

**Pull request merged**